integration-tests/bats/cherry-pick.bats

```diff
@@ -0,0 +1,439 @@
+
+    dolt checkout main
+
+    run dolt cherry-pick branch1~2
+    [ "$status" -eq "0" ]
+
+    run dolt sql -q "SELECT * FROM test" -r csv
+    [[ "$output" =~ "2,b" ]] || false
+    [[ ! "$output" =~ "1,a" ]] || false
+    [[ ! "$output" =~ "2,x" ]] || false
+    [[ ! "$output" =~ "3,c" ]] || false
+
+    run dolt cherry-pick branch1
+    [ "$status" -eq "0" ]
+
+    run dolt sql -q "SELECT * FROM test" -r csv
+    [[ ! "$output" =~ "1,a" ]] || false
+    [[ "$output" =~ "2,x" ]] || false
+    [[ ! "$output" =~ "3,c" ]] || false
+    [[ "$output" =~ "5,g" ]] || false
+    [[ "$output" =~ "8,u" ]] || false
+}
+
+@test "cherry-pick: too far back" {
+    dolt checkout main
+    run dolt cherry-pick branch1~10
+    [ "$status" -eq "1" ]
+    [[ "$output" =~ "ancestor" ]] || false
+}
+
+@test "cherry-pick: no changes" {
+    dolt commit --allow-empty -m "empty commit"
+    dolt checkout main
+    run dolt cherry-pick branch1
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ "No changes were made" ]] || false
+}
+
+@test "cherry-pick: invalid hash" {
+    run dolt cherry-pick aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+    [ "$status" -eq "1" ]
+    [[ "$output" =~ "target commit not found" ]] || false
+}
+
+@test "cherry-pick: has changes in the working set" {
+    dolt checkout main
+    dolt sql -q "INSERT INTO test VALUES (4, 'f')"
+    run dolt cherry-pick branch1~2
+    [ "$status" -eq "1" ]
+    [[ "$output" =~ "changes" ]] || false
+}
+
+@test "cherry-pick: update and delete on non existent row" {
+    run dolt sql -q "SELECT * FROM test" -r csv
+    [[ "$output" =~ "1,a" ]] || false
+    [[ "$output" =~ "2,b" ]] || false
+    [[ "$output" =~ "3,c" ]] || false
+
+    dolt sql -q "UPDATE test SET v = 'x' WHERE pk = 2"
+    dolt sql -q "DELETE FROM test WHERE pk = 1"
+    dolt add -A
+    dolt commit -m "Update and delete rows"
+
+    dolt checkout main
+
+    run dolt cherry-pick branch1
+    [ "$status" -eq "0" ]
+
+    run dolt sql -q "SELECT * FROM test" -r csv
+    [[ ! "$output" =~ "1,a" ]] || false
+    [[ ! "$output" =~ "2,b" ]] || false
+    [[ ! "$output" =~ "2,x" ]] || false
+    [[ ! "$output" =~ "3,c" ]] || false
+}
+
+@test "cherry-pick: insert duplicate rows" {
+    run dolt sql -q "SELECT * FROM test" -r csv
+    [[ "$output" =~ "1,a" ]] || false
+    [[ "$output" =~ "2,b" ]] || false
+    [[ "$output" =~ "3,c" ]] || false
+
+    dolt checkout main
+
+    dolt sql -q "INSERT INTO test VALUES (3, 'c')"
+    dolt add -A
+    dolt commit -m "Inserted 3 on main"
+
+    run dolt cherry-pick branch1
+    [ "$status" -eq "1" ]
+    [[ "$output" =~ "duplicate key error" ]] || false
+}
+
+@test "cherry-pick: insert, update, delete rows and schema changes on non existent table in working set" {
+    dolt sql -q "CREATE TABLE branch1table (id int primary key, col1 int)"
+    dolt sql -q "INSERT INTO branch1table VALUES (9,8),(7,6),(5,4)"
+    dolt add -A
+    dolt commit -m "create table with rows"
```

Comment:
(minor) consider just using `dolt commit -am "<message>"` to shorten this test file. If that pattern appears 10 times, it's worth it to save those 10 lines of test code.